assets/js/modules/analytics/common/timzezone-select.js

```diff
@@ -0,0 +1,133 @@
+	Select,
+	Option,
+} from '../../../material-components';
+import { countries } from './countries';
+import classnames from 'classnames';
+
+const { default: { country: allCountries } } = countries;
+
+const TimezoneSelect = ( { timezone, setTimezone, hasError } ) => {
+	const [ selectedCountry, setSelectedCountry ] = useState( timezone );
+	const [ selectedTimezoneID, setSelectedTimezoneID ] = useState( '' );
+	let multiTimezone,
+		selectedTimezoneDisplay = selectedTimezoneID;
+
+	const getTimezoneSelector = () => {
+		const response = (
+			<div >
+				<span>
+					<Select
+						className={ classnames(
+							'googlesitekit-analytics__select-timezone first',
+							{ 'mdc-text-field--error': hasError }
+						) }
+						name="country"
+						enhanced
+						value={ selectedCountry }
+						onEnhancedChange={ ( i, item ) => {
+							setTimezone( item.dataset.value );
+							setSelectedCountry( item.dataset.value );
+						} }
+						label={ __( 'Country', 'google-site-kit' ) }
+						outlined
+					>
+						{
+							allCountries && allCountries
+								.map( ( aCountry ) => {
+									// If the selected timezone is in this country, the country should be selected.
+									let value = aCountry.defaultTimeZoneId;
+									const timezoneMatch = aCountry.timeZone.find( ( tz ) => tz.timeZoneId === timezone );
+									if ( timezoneMatch ) {
+										value = timezoneMatch.timeZoneId;
+										if ( aCountry.timeZone.length > 1 ) {
+											multiTimezone = aCountry.timeZone;
+										} else {
+											setSelectedTimezoneID( aCountry.timeZone[ 0 ].displayName );
+											selectedTimezoneDisplay = aCountry.timeZone[ 0 ].displayName;
+											multiTimezone = false;
+										}
+										setSelectedCountry( timezoneMatch.timeZoneId );
+									}
+
+									return (
+										<Option
+											key={ aCountry.displayName }
+											value={ value }
+										>
+											{ aCountry.displayName }
+										</Option>
+									);
+								} ) }
+					</Select>
+				</span>
+				<span>
+					{ multiTimezone
+						? <Select
+							className="googlesitekit-analytics__select-timezone"
+							name="timezone2"
+							style={ { minWidth: '240px' } /*todo: move to css */ }
+							enhanced
+							value={ timezone }
+							onEnhancedChange={ ( i, item ) => {
+								setTimezone( item.dataset.value );
+							} }
+							label={ __( 'Timezone', 'google-site-kit' ) }
+							outlined
+						>
+							{
+								multiTimezone
+									.map( ( aTimezone ) => {
+										return (
+											<Option
+												key={ aTimezone.displayName }
+												value={ aTimezone.timeZoneId }
+											>
+												{ aTimezone.displayName }
+											</Option>
+										);
+									} )
+							}
+						</Select>
+						: selectedTimezoneDisplay
+					}
+				</span>
+			</div>
+		);
+
+		return response;
+	};
+
+	const timezoneSelector = useMemo( () => getTimezoneSelector(), [ selectedCountry, timezone, hasError ] );
```

Comment:
Is this expensive to render? It seems unnecessary to memoize.

Reply:
Yes, it is expensive due to the large dataset and multiple level array. Without the memoization typing into other fields is noticeably slow

Reply:
Hmm, I'll try to see if there are any easy perf-wins here but otherwise, let's at least add a comment as to why this is used for now 👍

Reply:
Added a comment.

Reply:
See above - I don't think it's complex to render, but it's setting state above multiple times while iterating over the options which is likely the reason it is noticeably slower without the memo.

Reply:
see below: https://github.com/google/site-kit-wp/pull/1389#discussion_r420455380